### ISSUE TITLE
Datepicket can't be navigated with keyboad

### ISSUE THREAD

**Issue by mlmoravek** [oruga-ui/oruga#1420]
### Overview of the problem

**Oruga** version: [0.11.6]

### Description

Currently, the Datepicker component cannot be navigated using keyboard navigation because all keyboard inputs are handled by the Dropdown wrapper component. 

As the Datepicker component contains a single complex widget rather than a list of items, normal Dropdown logic shouldn't apply to it.

### Expected behavior

In addition to the current dropdown implementation, the dropdown component can be configured to handle complex widget elements rather than a list of items. 

Once the focus is inside the dropdown menu, the menu's elements can be navigated using keyboard interactions.

### Actual behavior

Datepicker component can't be navigated with keyboad navigation.
